File: javascript/src/Structural/Adapter/TeaBag.js
(function() {
  define([], function() {
    var TeaBag;
    TeaBag = (function() {
      TeaBag.prototype.teaBagIsSteeped = null;

      function TeaBag() {
        this.teaBagIsSteeped = false;
      }

      TeaBag.prototype.steepTeaInCup = function() {
        this.teaBagIsSteeped = true;
        return 'tea bag is steeping in cup';
      };

      return TeaBag;

    })();
    return TeaBag;
  });

}).call(this);

File: ruby/src/Structural/Adapter/TeaBag.rb

class TeaBag

	def initialize()
		@teaBagIsSteeped = false
	end

	def steepTeaInCup()
 		@teaBagIsSteeped = true
		return 'tea bag is steeping in cup'
	end

	def getTeaBagIsSteeped
		@teaBagIsSteeped
	end
end

